src/vcpkg/commands.depend-info.cpp

```diff
@@ -88,59 +88,63 @@ namespace
-            Checks::unreachable(VCPKG_LINE_INFO, fmt::format("Not found in dependency graph: {}", package));
-        }
-
-        PackageDependInfo& info = iter->second;
-
-        if (depth > info.depth)
-        {
-            info.depth = depth;
-            if (depth < max_depth)
+            int new_depth = it->depth + 1;
+            for (const auto& dependency : it->dependencies)
             {
-                for (auto&& dependency : info.dependencies)
+                auto match = dependencies_map.find(dependency);
+                if (match == dependencies_map.end())
                 {
-                    assign_depth_to_dependencies(dependency, depth + 1, max_depth, dependencies_map);
+                    Checks::unreachable(VCPKG_LINE_INFO, fmt::format("Not found in dependency graph: {}", dependency));
+                }
+
+                if (match->second.depth < new_depth)
+                {
+                    match->second.depth = new_depth;
                 }
             }
-        }
+        };
     }
 
     std::vector<PackageDependInfo> extract_depend_info(const std::vector<const InstallPlanAction*>& install_actions,
+                                                       const Triplet& default_triplet,
+                                                       const Triplet& host_triplet,
                                                        const int max_depth)
     {
-        std::map<std::string, PackageDependInfo> package_dependencies;
+        const bool is_native = default_triplet == host_triplet;
+        auto decorated_name = [&default_triplet, &host_triplet, is_native](const PackageSpec& spec) -> std::string {
+            if (!is_native && spec.triplet() == host_triplet) return spec.name() + ":host";
+            if (spec.triplet() == default_triplet) return spec.name();
+            return spec.name() + ':' + spec.triplet().canonical_name();
+        };
+
+        std::vector<PackageDependInfo> out;
+        out.reserve(install_actions.size());
+
+        std::map<std::string, PackageDependInfo&> package_dependencies;
         for (const InstallPlanAction* pia : install_actions)
         {
             const InstallPlanAction& install_action = *pia;
 
             const std::vector<std::string> dependencies =
-                Util::fmap(install_action.package_dependencies, [](const PackageSpec& spec) { return spec.name(); });
+                Util::fmap(install_action.package_dependencies,
+                           [&decorated_name](const PackageSpec& spec) { return decorated_name(spec); });
 
             std::unordered_set<std::string> features{install_action.feature_list.begin(),
                                                      install_action.feature_list.end()};
             features.erase(FeatureNameCore.to_string());
 
-            auto& port_name = install_action.spec.name();
+            out.push_back({decorated_name(install_action.spec), 0, std::move(features), std::move(dependencies)});
```

Comment:
Using emplace_back here would allow us to get rid of the calls to back below.

Reply:
It is good enough.